starter-code/main.js

```diff
@@ -0,0 +1,237 @@
+                    } else {
+                        console.log(`${result[0].name} has ${result[0].number_of_employees} employees`);
+                        rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    }
+                    })
+                break;
+            case "10":
+            db.collection('companies').find({name: "Facebook"}, {name: 1, products:1, _id: 0}).toArray((error, result) => {
+                if (error) {
+                    console.log(error);
+                    rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                } else {
+                    console.log(result[0].products);
+                    rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                }
+                })
+                break;
+
+////////////////////////////////LO HE RESUELTO PERO NI IDEA DE POR QUÉ NO CONSEGUÍA ACCEDER DENTRO DE LA PROPIEDAD UNDEFINED                 
+            case "11":
+                db.collection('companies').find({ $and: [{name: "Facebook"},{relationships: {$elemMatch:{is_past: false}}}]}, {relationships: 1, _id: 0}).toArray((error, result) => {
+                    if (error) {
+                        console.log(error);
+                        rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    } else {
+                        // Como objeto no hacía más que darme undefined, pero al pasarlo al array he conseguido trabajara con él. ¿Por qué? ni idea
+                        let relations = [...result[0].relationships] 
+                        relations.forEach( elm => console.log(elm.person));
+                        rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                    }
+                    })
+                    break;
+
+                case "12":
+                    db.collection('companies').find({ $and: [{name: "Facebook"},{relationships: {$elemMatch:{is_past: true}}}]}, {relationships: 1, _id: 0}).toArray((error, result) => {
+                        if (error) {
+                            console.log(error);
+                            rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                        } else {
+                            let relations = [...result[0].relationships] 
+                            console.log(`Ya no trabajan en Facebook ${relations.length} personas`);
+                            rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+                        }
+                        })
+                        break;
+                  
+                case "13":
+                    db.collection('companies').find( {relationships: {$elemMatch:{"person.first_name": "david-ebersman"}}}, {name: 1, _id: 0}).toArray((error, result) => {
+                        if (error) {
```

Comment:
`// David Ebersman has worked at the following firms, []`
Este resultado podría ajustar la lógica de la siguiente manera:
```
              .find(
                {
                  $and: [
                    { "relationships.person.first_name": "David" },
                    { "relationships.person.last_name": "Ebersman" },
                  ],
                },
```